blockstore.go

```diff
@@ -284,19 +317,29 @@ func (b *Blockstore) Get(cid cid.Cid) (blocks.Block, error) {
 	defer b.oplock.RUnlock()
 
 	var val []byte
+
+Retry:
 	err := b.env.View(func(txn *lmdb.Txn) error {
 		v, err := txn.Get(b.db, cid.Hash())
 		if err == nil {
 			val = v
 		}
 		return err
 	})
+
 	switch {
 	case err == nil:
 		return blocks.NewBlockWithCid(val, cid)
 	case lmdb.IsNotFound(err) || lmdb.IsErrno(err, lmdb.BadValSize):
 		// lmdb returns badvalsize with nil keys.
 		err = blockstore.ErrNotFound
+	case lmdb.IsErrno(err, lmdb.ReadersFull):
+		b.oplock.RUnlock() // yield.
+		d := b.jitteredDelay()
+		log.Warnf("get encountered MDB_READERS_FULL; waiting %s", d)
+		time.Sleep(d)
```

Comment:
Can we turn these 3 incantations into a function call? They are repeated several times.

Reply:
Yup.